server/src/internalClusterTest/java/org/opensearch/gateway/RecoveryFromGatewayIT.java

```diff
@@ -797,7 +798,9 @@ public void testBatchModeEnabledWithoutTimeout() throws Exception {
         );
         assertTrue(ExistingShardsAllocator.EXISTING_SHARDS_ALLOCATOR_BATCH_MODE.get(internalCluster().clusterService().getSettings()));
         assertEquals(1, gatewayAllocator.getNumberOfStartedShardBatches());
-        assertEquals(1, gatewayAllocator.getNumberOfStoreShardBatches());
+        // Replica shard would be marked ineligible since there are no data nodes.
+        // It would then be removed from any batch and batches would get deleted, so we would have 0 replica batches
```

Comment:
Can we add another testcase where we mark the batch as `fetching=true` before calling reroute ?

Reply:
This doesn't seem possible. I don't think we can mock the batch to force it to have `fetching=true` in an IT.

Reply:
I am trying to simulate blocking the `asyncFetch` in a similar manner to this test: 
https://github.com/rajiv-kv/OpenSearch/blob/7ffcd6500e0bd5956cef5c289ee66d9f99d533fc/server/src/internalClusterTest/java/org/opensearch/gateway/ReplicaShardAllocatorIT.java#L167

Will take this up as a follow up issue as it does not seem trivial to simulate this case